### PR TITLE
Fixing the short url logic

### PR DESCRIPTION
**Pull request by eldhosembabu**
Added back the regex to check for more than one path in short url.

### CODE REVIEW

File: FirebaseDynamicLinks/Sources/Utilities/FDLUtilities.m [hunk 1]
```diff
@@ -295,7 +295,10 @@ BOOL FIRDLMatchesShortLinkFormat(NSURL *URL) {
   // Path cannot be prefixed with /link/dismiss
   BOOL isDismiss = [[URL.path lowercaseString] hasPrefix:@"/link/dismiss"];
 
-  return hasPathOrCustomDomain && !isDismiss && canParse;
+  BOOL matchesRegularExpression =
+      ([URL.path rangeOfString:@"/[^/]+" options:NSRegularExpressionSearch].location != NSNotFound);
```
Comment (ncooke3) [inline FirebaseDynamicLinks/Sources/Utilities/FDLUtilities.m:299]: I saw `/[^/]+` pattern was used previously int he SDK. Do you think it is worth adding a comment for what this pattern means?

**Review by ncooke3 — APPROVED** [review]
Thanks for addressing this!

File: FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m [hunk 2]
```diff
@@ -1198,7 +1198,6 @@ - (void)testFailMatchesShortLinkFormatForCustomDomains {
 - (void)testPassMatchesShortLinkFormatForCustomDomains {
   NSArray<NSString *> *urlStrings = @[
     @"https://google.com/xyz", @"https://google.com/xyz/?link=https://somedomain",
-    @"https://google.com?link=https://somedomain", @"https://google.com/?link=https://somedomain",
```
Comment (paulb777) [inline FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m:1201]: From the description I thought we were missing a type of link.  Why is a test being deleted versus adding an additional pattern?

Comment (eldhosembabu) [reply FirebaseDynamicLinks/Tests/Unit/FIRDynamicLinksTest.m:1201]: The link that we are testing here is a valid FDL link, but not a short link. I added this one when i did re writing the logic but its actually a long link. When I added back the missed regex validation, this test case is caught. So removing it now.

File: FirebaseDynamicLinks/Sources/Utilities/FDLUtilities.m (hunk 1 above)
Comment (eldhosembabu) [reply FirebaseDynamicLinks/Sources/Utilities/FDLUtilities.m:299]: Based on my understanding, this regex is checking for a "/" and should not be having anymore succeeding "/"'s. I'll add it as a comment there.

Comment (eldhosembabu) [reply FirebaseDynamicLinks/Sources/Utilities/FDLUtilities.m:300]: Done

**Review by paulb777 — APPROVED** [review]

**Review by ncooke3 — APPROVED** [review]

**Pull request merged**